pkg/config/config.go

```diff
@@ -160,6 +168,41 @@ func LoadFile(filename string, expandEnvVars bool, c *Config) error {
 	return LoadBytes(buf, expandEnvVars, c)
 }
 
+// LoadRemote reads a config from url
+func LoadRemote(url string, expandEnvVars bool, c *Config) error {
+	remoteOpts := &RemoteOpts{}
+	if c.BasicAuthUser != "" || c.BasicAuthPassFile != "" {
+		remoteOpts.HTTPClientConfig = &config.HTTPClientConfig{
+			BasicAuth: &config.BasicAuth{
+				Username:     c.BasicAuthUser,
+				PasswordFile: c.BasicAuthPassFile,
+			},
+		}
+	}
```

Comment:
does basic auth work if you just have a password but no username? 🤔

Reply:
no I don't think so. this condition should be `&&` instead of `||`